### ISSUE TITLE
Masked `threshold_otsu`

### ISSUE THREAD

**Issue by PeterJackNaylor** [scikit-image/scikit-image#3604]
I was wondering if the community would be interested in a modified version of threshold_otsu to incorporate a mask.

I have already implemented this function with an extra keyword which was the mask. In particular, the threshold is computed only on the pixels where the mask is equal to 1. Pixels that are not in the mask will be clustered with the background pixels. It involves adding an argument and changing one line to take into account the mask when we unravel the values in the image.
'' raveled_image = image[mask > 0].ravel() ''

I thought this was interesting to have in the situation of partial labelling and where an ordinnary threshold_otsu would fail due to the presence of unwanted artefacts. 
For example: in my situations I wanted to find tissue on glass plates but doctors write on the sides of the tissue (which is super annoying) and this additionnal writting messes up the histogram of values, so the mask was a way to prevent that.

**Comment by soupault** [comment]
@PeterJackNaylor Absolutely, yes! Support for masked images in not yet common in scikit-image, but has been identified as one of the directions where we'd like to go.
Please, see the thread in https://github.com/scikit-image/scikit-image/issues/1624 for more info regarding the conventions.

**Comment by mkcor** [comment]
@PeterJackNaylor any chance you could submit a PR for this (old) issue? :eyes:
